fl4health/datasets/skin_cancer/load_data.py

```diff
@@ -0,0 +1,204 @@
+# The following code is adapted from the medical_federated GitHub repository by Seongjun Yang et al.
+# Paper: https://arxiv.org/abs/2207.03075
+# Code: https://github.com/wns823/medical_federated.git
+# - medical_federated/skin_cancer_federated/split_dataset.py
+# - medical_federated/skin_cancer_federated/skin_cancer_datasets.py
+
+import json
+import random
+from concurrent.futures import ThreadPoolExecutor
+from logging import ERROR, INFO
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+import torchvision.transforms as transforms
+from flwr.common.logger import log
+from PIL import Image
+from torch.utils.data import DataLoader
+
+from fl4health.utils.dataset import TensorDataset
+from fl4health.utils.dataset_converter import DatasetConverter
+from fl4health.utils.sampler import LabelBasedSampler
+
+
+def construct_skin_cancer_tensor_dataset(
+    data: List[Dict[str, Any]], transform: Optional[Callable] = None, num_workers: int = 8
+) -> TensorDataset:
+    def load_image(item: Dict[str, Any]) -> Tuple[torch.Tensor, int]:
```

Comment:
Perhaps just pull this out into a separately defined function. Generally, I try to avoid locally defined functions unless really required and I think this one can be extracted easily.

Reply:
I see, I have pulled it out and added google docstring